From 588692667ae3d8a198760c19d335d33b0dca6ba0 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Thu, 1 Feb 2024 17:46:49 +0800
Subject: [PATCH 47/61] mtk: mac80211: fix radar required of link issue in
 reserve_reassign and reserve_assign

link->radar_required is not updated in
ieee80211_link_use_reserved_assign & ieee80211_link_use_reserved_reassign
This will lead to DFS RDD init incomplete (RDD_CAC_START, RDD_CAC_END &
RDD_DET_MODE is not set to fw)

CR-Id: WCNCR00274293
Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Change-Id: I44eeb32be5ff42202639f94418d079930369fde9
---
 net/mac80211/chan.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/mac80211/chan.c b/net/mac80211/chan.c
index 8043d1d..ac22524 100644
--- a/net/mac80211/chan.c
+++ b/net/mac80211/chan.c
@@ -1207,6 +1207,7 @@ ieee80211_link_use_reserved_reassign(struct ieee80211_link_data *link)
 	if (link_conf->chanreq.oper.width != link->reserved.oper.width)
 		changed = BSS_CHANGED_BANDWIDTH;
 
+	link->radar_required = link->reserved_radar_required;
 	ieee80211_link_update_chanreq(link, &link->reserved);
 
 	_ieee80211_change_chanctx(local, new_ctx, old_ctx, chanreq, link);
@@ -1290,6 +1291,7 @@ ieee80211_link_use_reserved_assign(struct ieee80211_link_data *link)
 	list_del(&link->reserved_chanctx_list);
 	link->reserved_chanctx = NULL;
 
+	link->radar_required = link->reserved_radar_required;
 	err = ieee80211_assign_link_chanctx(link, new_ctx);
 	if (err) {
 		if (ieee80211_chanctx_refcount(local, new_ctx) == 0)
-- 
2.39.2

